From 08632c616e02dbe75a82b47794715ea4360a5b9e Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Fri, 19 Dec 2025 14:00:24 +0100
Subject: [PATCH] usb: dwc3: core: disable platform lowlevel hw resources
 during shutdown

Disable platform lowlevel hw resources such as phy, clock and
regulators in device shutdown hook. It will allow to properly
probe the driver on next warm reboot.

Change-Id: I071ff8aa2d14df0063b745bb5667b43944f60a11
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/519769
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 drivers/usb/dwc3/core.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -2453,6 +2453,7 @@ MODULE_DEVICE_TABLE(acpi, dwc3_acpi_matc
 static struct platform_driver dwc3_driver = {
 	.probe		= dwc3_probe,
 	.remove_new	= dwc3_remove,
+	.shutdown	= dwc3_remove,
 	.driver		= {
 		.name	= "dwc3",
 		.of_match_table	= of_match_ptr(of_dwc3_match),
